### ISSUE TITLE
Insanely large RAM usage when spawning a lot of entities

### ISSUE THREAD

**Issue by ELginas** [feather-rs/feather#441]
## Description

Feather uses way too much RAM for handling entities. When spawning 10 000 piglin brutes 1 in between each other, game uses 2 GB+ RAM and usage changes rapidly. RAM spike only happens when you are connected to the server.

## Reproduction Steps

1. Modify query_entities plugin so that piglin brutes spawn every 1 block instead of every 12 blocks.
2. Add modified query_entities plugin
3. Launch feather server
4. Login to feather server and see RAM go up A LOT

## What You Expected to Happen

I expected maybe increased CPU usage when joining server with a lot of entities and I also expected RAM to go up slightly because of loaded chunks.

## What Actually Happened

After I joined the server CPU usage went up by 20% and RAM usage went up by ~2 GB and fluctuated uncontrollably (by about ~400 MB every second).

## Your Environment

- Operating System: **Windows 10**
- Operating System Version: **21H1**
- **Feather** Version: **latest**

## Additional Context

System RAM: **8 GB**
Minecraft client: **Lunar**
View distance: **8 chunks** (but Lunar tends to load chunks more slowly to maintain good FPS)

When no one is on the server, CPU usage stays around ~20% and RAM usage is ~6 MB.
Also when I joined server, my CPU reached 100% and server printed `Tick took too long` once in a while and my RAM stayed around ~95% usage.

**Comment by amberkowalski** [comment]
Can't reproduce, this might be an issue with debug mode or something. (Or maybe I did something wrong hehe)

![image](https://user-images.githubusercontent.com/4877041/124382963-51465380-dcca-11eb-97a9-d1f4c9b85a5d.png)

Ignore CPU usage that's just because I have a garbage CPU right now.

[Memory Usage Information (Generated with VMMap](https://cdn.discordapp.com/attachments/805342116057579520/861204566519054336/feather-server-release-10k.mmp)

As shown by the above, it only got to 100/80 MB of memory use.

**Comment by ELginas** [comment]
I tried with Vanilla 1.16.4 client, RAM and CPU still rises by a lot, but RAM usage is lower because client is having hard time loading chunks and handling entities. Also, did you join the server once you started the server? Added plugin for testing.
[query-entities.zip](https://github.com/feather-rs/feather/files/6759830/query-entities.zip)

**Comment by caelunshun** [comment]
> I tried with Vanilla 1.16.4 client, RAM and CPU still rises by a lot, but RAM usage is lower because client is having hard time loading chunks and handling entities. Also, did you join the server once you started the server? Added plugin for testing.
> [query-entities.zip](https://github.com/feather-rs/feather/files/6759830/query-entities.zip)

Are you measuring the RAM usage of the server itself or on your whole system? It makes sense for the client to use that much RAM with 10,000 entities, and the "fluctuating" pattern sounds like the Java GC.

**Comment by ELginas** [comment]
No, I'm talking about the Feather server. Of course client will have awful time near 10,000 entities😄
